src/main/java/com/google/cloud/resourcemanager/ConstraintInfo.java

```diff
@@ -0,0 +1,291 @@
+    }
+  }
+
+  ConstraintInfo(Builder builder) {
+    this.booleanConstraint = builder.booleanConstraint;
+    this.constraintDefault = builder.constraintDefault;
+    this.description = builder.description;
+    this.displayName = builder.displayName;
+    this.constraints = builder.constraints;
+    this.name = builder.name;
+    this.version = builder.version;
+  }
+
+  /** Returns the boolean constraint to check whether the constraint is enforced or not. */
+  public BooleanConstraint getBooleanConstraint() {
+    return booleanConstraint;
+  }
+
+  /** Returns the default behavior of the constraint. */
+  public String getConstraintDefault() {
+    return constraintDefault;
+  }
+
+  /** Returns the detailed description of the constraint. */
+  public String getDescription() {
+    return description;
+  }
+
+  /** Returns the human readable name of the constraint. */
+  public String getDisplayName() {
+    return displayName;
+  }
+
+  /** Returns the listConstraintInfo. */
+  public Constraints getConstraints() {
+    return constraints;
+  }
+
+  /** Returns the globally unique name of the constraint. */
+  public String getName() {
+    return name;
+  }
+
+  /** Returns the version of the Constraint. Default version is 0. */
+  public Integer getVersion() {
+    return version;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    ConstraintInfo that = (ConstraintInfo) o;
+    return Objects.equals(booleanConstraint, that.booleanConstraint)
+        && Objects.equals(constraintDefault, that.constraintDefault)
+        && Objects.equals(description, that.description)
+        && Objects.equals(displayName, that.displayName)
+        && Objects.equals(constraints, that.constraints)
+        && Objects.equals(name, that.name)
+        && Objects.equals(version, that.version);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        booleanConstraint, constraintDefault, description, displayName, constraints, name, version);
+  }
+
+  /** Returns a builder for the {@link ConstraintInfo} object. */
+  public static Builder newBuilder(String name) {
+    return new Builder(name);
+  }
+
+  /** Returns a builder for the {@link ConstraintInfo} object. */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  Constraint toPb() {
+    Constraint constraintPb = new Constraint();
+    constraintPb.setBooleanConstraint(booleanConstraint);
+    constraintPb.setConstraintDefault(constraintDefault);
+    constraintPb.setDescription(description);
+    constraintPb.setDisplayName(displayName);
+    if (constraints != null) {
+      constraintPb.setListConstraint(constraints.toPb());
+    }
+    constraintPb.setName(name);
+    constraintPb.setVersion(version);
+    return constraintPb;
+  }
+
+  static ConstraintInfo fromPb(Constraint constraintPb) {
```

Comment:
fromProtobuf (which I guess is still an abbreviation but better)

Reply:
done